### PR TITLE
[Jetpack Focus] Revert Jetpack Notifications Disabling Feature Flag and Release Notes

### PR DESCRIPTION
**Pull request by staskus**
## Description

This PR that disables functionality of "Disable WordPress Notifications" project 

Reverting https://github.com/wordpress-mobile/WordPress-iOS/issues/19619

Internal ref: D94385-code

### Prerequisites

Confirm that `prevent_duplicate_notifs_remote_field":false` value exists in https://public-api.wordpress.com/wpcom/v2/mobile/feature-flags

### Case 1: Notifications not disabled on WordPress app
1. Install WordPress app and log in
2. Go to Notification tab on WordPress app and accept iOS notification permission
3. Install Jetpack app and log in
4. Go to Notification tab on Jetpackk app and accept iOS notification permission
5. Jetpack app should not redirect to WordPress app
6. Trigger push notification (write a comment on a blog)
7. Both Jetpack and WordPress apps should receive comments

### Case 2. No "Allow Notifications" switch should be visible on WordPress
1. Install WordPress app and log in
2. Go to Notification tab on WordPress app and accept iOS notification permission
3. Staying on Notifications tab, click Settings icon on a top right corner
4. Confirm that there's no "Allow Notifications" setting at the top of the settings view

## Regression Notes

1. Potential unintended areas of impact

The feature is isolated, there shouldn't be side-effects if we confirm that Jetpack app doesn't turn off WordPress notifications

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

Existing JetpackNotificationMigrationServiceTests continues running and succeeding, that test both enabled and disabled cases

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by twstokes** [comment]
Hey @staskus! 👋 

Could we add testing steps (and or screenshots) for the items under the "Testing" section so testers know how to get to those areas? Thanks! 🙇

**Comment by salimbraksa** [comment]
Hey @staskus, Jetpack app is still redirecting to WP when I allowed Notifications. Not sure if I'm doing something wrong:

| Recording |
| ----------- |
| <video src="https://user-images.githubusercontent.com/9609223/205745627-64295001-48b0-42ca-bebe-629abf6a1534.mp4" /> |

And all content migration feature flags were disabled:

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 21.3
 -----
+* [***] Adds a smooth, opt-in transition to the Jetpack app. [#19714]
```
Comment (twstokes) [inline RELEASE-NOTES.txt]: Looking into this. The `release/21.3` merge caused it to reappear, I think.

Comment (twstokes) [reply RELEASE-NOTES.txt]: This was resolved by a325d3d0d159650d1a6ddb39373f31e805e91696 which removed the re-addition of this line on the merge commit. I force-pushed it, so heads up @staskus and @salimbraksa. Thanks! 🙇

**Comment by twstokes** [comment]
> Hey @staskus, Jetpack app is still redirecting to WP when I allowed Notifications. Not sure if I'm doing something wrong:

I can confirm the same behavior running from the latest on this branch.


https://user-images.githubusercontent.com/2092798/205780319-c5e5bcd2-36e4-43dc-a8e7-8b1fcb75d741.mp4

**Comment by staskus** [comment]
@twstokes @salimbraksa it happened because the remote flag is still enabled and it overrides a local flag. Sorry for not mentioning it in the PR. There's a separate diff that disables it which is made but not yet deployed.

**Comment by staskus** [comment]
> Hey @staskus! 👋
> 
> Could we add testing steps (and or screenshots) for the items under the "Testing" section so testers know how to get to those areas? Thanks! 🙇

@twstokes okay. This was super quick PR before EOD prioritizing quick merge over the quality of description.

**Comment by staskus** [comment]
> Hey @staskus! 👋
> 
> Could we add testing steps (and or screenshots) for the items under the "Testing" section so testers know how to get to those areas? Thanks! 🙇

@twstokes Of course. This was super quick PR before EOD prioritizing quick merge over the quality of description.

**Comment by twstokes** [comment]
> @twstokes @salimbraksa it happened because the remote flag is still enabled and it overrides a local flag. Sorry for not mentioning it in the PR. There's a separate diff that disables it which is made but not yet deployed.

Ah, that makes sense! Thanks for the clarification.

> @twstokes okay. This was super quick PR before EOD prioritizing quick merge over the quality of description.

Totally understood. Thanks for the updates. 👍

**Comment by twstokes** [comment]
I've added the remote feature flag PR to the description of this one. I assume it will need to be merged before we can proceed with testing and merging this one.

**Update:** It was decided to override the remote functionality for this feature in `21.3`, so the API change has been cancelled.

Ref D94385-code

**Review by twstokes — APPROVED** [review]
LGTM @staskus. 🚀 

- Jetpack did not bring WordPress to the foreground when allowing notifications
- I received push notifications on both apps
- I did not see the "Allow Notifications" switch in WordPress

**Review by salimbraksa — APPROVED** [review]
I confirm this is working as expected. LGTM! 🚀

**Pull request merged**